src/main/java/com/karankumar/bookproject/ui/goal/SaveGoalEvent.java

```diff
@@ -0,0 +1,11 @@
+package com.karankumar.bookproject.ui.goal;
+
+import com.karankumar.bookproject.backend.entity.ReadingGoal;
+
+public class SaveGoalEvent {
+    public static class SaveEvent extends ComponentGoalEvent.GoalFormEvent {
```

Comment:
As with `DeleteEvent`, could you please remove the inner class here and make the outer class extend `ComponentGoalEvent.GoalFormEvent`?

Reply:
I think you forgot to do this, so I'll add this in